src/freedesktop/detect.rs

```diff
@@ -0,0 +1,64 @@
+use anyhow::Context;
+use detect_desktop_environment::DesktopEnvironment;
+use ini::Ini;
+use std::path::{Path, PathBuf};
+
+use crate::Mode;
+
+use super::get_freedesktop_color_scheme;
+
+const XDG_KDEGLOBALS: &str = "/etc/xdg/kdeglobals";
+
+fn detect_gtk(pattern: &str) -> Mode {
+    match dconf_rs::get_string(pattern) {
+        Ok(theme) => Mode::from(theme.to_lowercase().contains("dark")),
+        Err(_) => Mode::Default,
+    }
+}
+
+fn detect_kde(path: &str) -> anyhow::Result<Mode> {
+    let cfg = Ini::load_from_file(path)?;
+    let section = cfg.section(Some("Colors:Window")).with_context(|| "Failed to get section Colors:Window")?;
+    let values = section.get("BackgroundNormal").with_context(|| "Failed to get BackgroundNormal inside Colors:Window")?;
+    let rgb = values
+        .split(',')
+        .map(|s| s.parse::<u32>().unwrap_or(255))
+        .collect::<Vec<u32>>();
+    let rgb = if rgb.len() > 2 {
```

Comment:
```suggestion
    let rgb = if rgb.len() >= 3 {
```
I think this conveys the intent a bit more clearly.

Reply:
Done! 😸